SimPEG/EM/Analytics/NSEM.py

```diff
@@ -0,0 +1,141 @@
+        epsmodel = mu_r
+
+    sigcm = np.zeros_like(sigmodel, dtype='complex_')
+    if chg == 0. or taux == 0. or c == 0.:
+        sigcm = sigmodel
+    else:
+        for j in range(1, len(sigcm)):
+            sigcm[j] = _PCC(sigmodel[j], chg[j], taux[j], c[j], f)
+
+    sigcm = np.append(np.r_[0.], sigcm)
+    mu = np.append(np.r_[1.], mumodel)*mu_0
+    eps = np.append(np.r_[1.], epsmodel)*epsilon_0
+    H = np.append(np.r_[1.2*(1e5)], thickness)
+
+    K = k(f, sigcm, mu, eps)
+    Z = _ImpZ(f, mu, K)
+
+    EH = np.matrix(np.zeros((2, n+1), dtype='complex_'), dtype='complex_')
+    UD = np.matrix(np.zeros((2, n+1), dtype='complex_'), dtype='complex_')
+
+    UD[1, -1] = 1.
+
+    for i in range(-2, -(n+2), -1):
+
+        UD[:, i] = _Tinv(H[i+1], K[i])*_Pinv(Z[i])*_P(Z[i+1])*UD[:, i+1]
+        UD = UD/((np.abs(UD[0, :]+UD[1, :])).max())
+
+    for j in range(0, n+1):
+        EH[:, j] = np.matrix([[1., 1, ], [-1./Z[j], 1./Z[j]]])*UD[:, j]
+
+    return UD, EH, Z, K
+
+# Utils to compute the apparent impedance over a layered Earth Model
+def MT_LayeredEarth(freq, thickness, sig, nlayer, return_type='Res-Phase', chg=0., tau=0., c=0., mu_r=1., eps_r=1.):
+    """
+    This code compute the analytic response of a n-layered Earth to a plane wave (Magnetotellurics).
+    The solution is first developed in Ward and Hohmann 1988.
+    See also http://em.geosci.xyz/content/maxwell3_fdem/natural_sources/MT_N_layered_Earth.html
+
+    :param freq: the frequency at which we take the measurements
+    :type freq: float or numpy.array
+    :param thickness: thickness of the Earth layers in meters
+    :type thickness: float or numpy.array
+    :param sig: electric conductivity of the Earth layers in S/m
+    :type sig: float or numpy.array
+    :param int nlayer: number of Earth layer
+    :param str return_type: Output return_type. 'Res-Phase' returns apparent resisitivity and Phase. 'Complex' returns the complex Impedance
+    :param numpy.array chg: Cole-Cole Parameters for chargeable layers: chargeability
+    :param numpy.array tau: Cole-Cole Parameters for chargeable layers: time decay constant
+    :param numpy.array c: Cole-Cole Parameters for chargeable layers: geometric factor
+    :param mu_r: relative magnetic permeability
+    :type mu_r: float or numpy.array
+    :param eps_r: relative dielectric permittivity
+    :type eps_r: float or numpy.array
+    """
+
+    if isinstance(freq, float):
+        F = np.r_[freq]
+    else:
+        F = freq
+
+    Res = np.zeros_like(F)
+    Phase = np.zeros_like(F)
+    App_ImpZ = np.zeros_like(F, dtype='complex_')
+
+    for i in range(0, len(F)):
+        _, EH, _, _ = _Propagate(F[i], thickness, sig, chg, tau, c, mu_r, eps_r, nlayer)
+
+        App_ImpZ[i] = EH[0, 1]/EH[1, 1]
+
+        Res[i] = np.abs(App_ImpZ[i])**2./(mu_0*omega(F[i]))
+        Phase[i] = np.angle(App_ImpZ[i], deg=True)
+
+    if return_type == 'Res-Phase':
+        return Res, Phase
+
+    elif return_type == 'Complex':
```

Comment:
how about labeling this "Impedance" ? (Complex doesn't fully capture what it is). Thoughts @thast?